sasl/sasl.go

```diff
@@ -42,3 +44,21 @@ type StateMachine interface {
 	// value will be true.
 	Next(ctx context.Context, challenge []byte) (done bool, response []byte, err error)
 }
+
+// Metadata contains additional data for performing SASL authentication.
+type Metadata struct {
+	// BrokerAddress is the address of the broker the authentication will be
+	// performed on.
```

Comment:
Is this address meant to be the logical hostname or the resolved network address (IP+port)?

Reply:
It's meant to be the logical hostname. Its host part is then used to generate the SPN (Service Principal Name) in the context of the GSSAPI authentication.